### PR TITLE
Add missing labels to resources managed by the operator

### LINKED ISSUE #799: common label quarks.cloudfoundry.org/deployment-name across all dm, ig, runtime pods, and secrets, etc
I'd like to be able to use `stern -l quarks.cloudfoundry.org/deployment-name=sample-webapp-using-redis` to see all logs from all the pods used to prepare + run my deployment. Currently I think this label is only applied to the runtime pods, not to the preparatory pods like `ig`:

Labels from an `ig` pod:

```
Labels:         controller-uid=70da82ba-fa74-4718-a901-a0b4ea2ef912
                delete=pod
                job-name=ig-sample-webapp-using-redis-15e3e1984d702274
                quarks.cloudfoundry.org/qjob-name=ig-sample-webapp-using-redis
```

And from a resulting running pod from the same quarks deployment:

```
Labels:         controller-revision-hash=sample-webapp-using-redis-redis-bb6d548c5
                quarks.cloudfoundry.org/az-index=0
                quarks.cloudfoundry.org/deployment-name=sample-webapp-using-redis
                quarks.cloudfoundry.org/instance-group-name=redis
                quarks.cloudfoundry.org/pod-ordinal=0
                quarks.cloudfoundry.org/quarks-statefulset-name=sample-webapp-using-redis-redis
                statefulset.kubernetes.io/pod-name=sample-webapp-using-redis-redis-0
```

Could we add `quarks.cloudfoundry.org/deployment-name` to all the pods for that deployment?

Similarly, it looks like generated secrets don't have this label either:

Secret `sample-webapp-using-redis.var-redis-password` labels:

```
  labels:
    quarks.cloudfoundry.org/secret-kind: generated
```

I trust there's nothing wrong with wanting more labels on more things, so I'll investigate a PR.

### PR DESCRIPTION
**Pull request by cibaca**
[171010810](https://www.pivotaltracker.com/n/projects/2192232/stories/171010810)

fixes #799

### CODE REVIEW

File: pkg/kube/controllers/quarkssecret/quarkssecret_reconciler.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"code.cloudfoundry.org/cf-operator/pkg/credsgen"
+	bdv1 "code.cloudfoundry.org/cf-operator/pkg/kube/apis/boshdeployment/v1alpha1"
```
Comment (manno) [inline pkg/kube/controllers/quarkssecret/quarkssecret_reconciler.go]: We have a story in the ice box to make quarks secret a stand alone helm chart. In that case, we wouldn't want it to have any knowledge of the bdpl (boshdeployment) CRD.

I'd suggest to not use `bdv1.LabelDeploymentName` here, but instead copy a set of generic labels to the secret.
QuarksJob has the same problem: https://github.com/cloudfoundry-incubator/quarks-job/blob/594102fffff0033a3f5040a66d1b9e0c10af9906/pkg/kube/apis/quarksjob/v1alpha1/types.go#L88

**Comment by viovanov** [comment]
@cibaca can you please resolve this?

**Comment by viovanov** [comment]
@manno can you please review that your comment was addressed?

File: pkg/kube/controllers/quarkssecret/quarkssecret_reconciler.go [hunk 2]
```diff
@@ -100,10 +100,10 @@ func (r *ReconcileQuarksSecret) Reconcile(request reconcile.Request) (reconcile.
 		ctxlog.Info(ctx, "Error reading the object")
 		return reconcile.Result{}, errors.Wrap(err, "Error reading quarksSecret")
 	}
-
 	if meltdown.NewWindow(r.config.MeltdownDuration, qsec.Status.LastReconcile).Contains(time.Now()) {
 		ctxlog.WithEvent(qsec, "Meltdown").Debugf(ctx, "Resource '%s' is in meltdown, requeue reconcile after %s", qsec.GetNamespacedName(), r.config.MeltdownRequeueAfter)
 		return reconcile.Result{RequeueAfter: r.config.MeltdownRequeueAfter}, nil
+
```
Comment (manno) [inline pkg/kube/controllers/quarkssecret/quarkssecret_reconciler.go:106]: nit: newline

File: pkg/bosh/qjobs/job_factory_test.go [hunk 3]
```diff
@@ -164,6 +164,12 @@ var _ = Describe("JobFactory", func() {
 			Expect(len(spec.InitContainers)).To(BeNumerically("<", 2))
 			Expect(len(spec.Containers)).To(BeNumerically("<", 2))
 		})
+		It("should have required labels", func() {
```
Comment (manno) [inline pkg/bosh/qjobs/job_factory_test.go:167]: nit: newline between it blocks missing :)

**Pull request merged**